_stbt/core.py

```diff
@@ -2491,24 +2527,32 @@ def _iter_frames(depth=1):
 def _find_path(filename):
     """Searches for the given filename and returns the full path.
 
-    Searches in the directory of the script that called (for example)
-    detect_match, then in the directory of that script's caller, etc.
+    Searches in the directory of the script that called `load_image` (or
+    `match`, etc), then in the directory of that script's caller, etc.
     """
 
     if os.path.isabs(filename):
         return filename
 
-    # stack()[0] is _find_path;
-    # stack()[1] is _find_path's caller, e.g. detect_match;
-    # stack()[2] is detect_match's caller (the user script).
+    # Start searching from the first parent stack-frame that is outside of
+    # the _stbt installation directory (this file's directory). We can ignore
+    # the first 2 stack-frames:
+    #
+    # * stack()[0] is _find_path;
+    # * stack()[1] is _find_path's caller: load_image or _load_template;
+    # * stack()[2] is load_image's caller (the user script). It could also be
+    #   _load_template's caller (e.g. `match`) so we still need to check until
+    #   we're outside of the _stbt directory.
```

Comment:
I'm not a huge fan of the way we do image lookup.  It makes caching difficult and the result of calling match in a function non-deterministic.

Some alternatives:

* Use a similar trick to pytest which has a [`__tracebackhide__`](https://docs.pytest.org/en/latest/example/simple.html#writing-well-integrated-assertion-helpers) attribute for helpers.  We would then consider only the first stack frame that isn't marked with `__tracebackhide__`.
* Expect to find all images under some prefix in the test-pack - this would bind our implementation to the concept of a test-pack though.

Oh well, it's too late now, we have to preserve the algorithm as it is to avoid breaking test-scripts.